src/metabase/lib/aggregation.cljc

```diff
@@ -0,0 +1,35 @@
+(ns metabase.lib.aggregation
+  (:refer-clojure :exclude [count distinct max min])
+  (:require
+   [metabase.lib.common :as lib.common]
+   [metabase.lib.options :as lib.options]
+   [metabase.util.malli :as mu])
+  #?(:cljs (:require-macros [metabase.lib.aggregation])))
+
+(mu/defn count :- [:or
+                   fn?
+                   :mbql.clause/count]
+  "Create a `count` filter clause."
+  ([]
+   #_{:clj-kondo/ignore [:redundant-fn-wrapper]}
+   (fn [query stage-number]
+     (count query stage-number)))
+  ([x]
+   (fn [query stage-number]
+     (count query stage-number x)))
```

Comment:
I would prefer having these in the dev namespace. As I understand these functions are for use in the REPL. That would help sharpening the return type.

Reply:
Hmm, we have to figure out how you're actually supposed to use these functions

I thought functions like `sum` would be the functions that adds a `:count` aggregation to a query, e.g.

```clj
(-> query
      (sum <some-field>))
```

And we'd have a separate set of functions like `sum-clause` to create the clauses separately, then you could add them later

e.g. 

```clj
(sum-clause ...)
```

I thought that's what we talked about in Slack at least. 

We need to figure out how actual usage of this will look to add simple aggregations like `sum` and to add compound aggregations like `[:+ [:sum X] [:sum Y]]`... 

I think Tamas is right and maybe we can have different more permissive versions of these functions that handle and optionally return functions in `lib.core`, since that functionality would presumably only be used in REPL and test situations. In FE usage I don't think we should have unresolved stuff anywhere